app/services/trainees/create_from_apply.rb

```diff
@@ -150,10 +150,41 @@ def nationality_names
     end
 
     def trainee_already_exists?
-      scope = application_record.provider.trainees.not_withdrawn.or(Trainee.not_awarded)
-      scope.exists?(first_names: raw_trainee["first_name"],
-                    last_name: raw_trainee["last_name"],
-                    date_of_birth: raw_trainee["date_of_birth"])
+      potential_duplicates.any? { |trainee| confirmed_duplicate?(trainee) }
```

Comment:
what you think of pulling out all the duplicate checking into its own class?

Reply:
Good point. FWIW I think it's even more helpful for the test code.